include/sys/interrupt.h

```diff
@@ -75,11 +86,19 @@ struct intr_handler {
   void *ih_argument;        /* argument to pass to filter/service routines */
   const char *ih_name;      /* name of the handler */
   /* XXX: do we really need ih_prio? it has no real use cases so far... */
-  prio_t ih_prio; /* handler's priority (sort key for ie_handlers) */
+  prio_t ih_prio;   /* handler's priority (sort key for ie_handlers) */
+  uint8_t ih_flags; /* flags (currently if to be removed or delegated to a
+                       private thread) */
 };
 
 typedef TAILQ_HEAD(, intr_handler) ih_list_t;
 
+typedef struct intr_thread {
+  intr_event_t *it_event; /* Associated event */
+  thread_t *it_thread;    /* Kernel thread. */
+  ih_list_t it_wchan;
```

Comment:
Why do we need this field? This is used only as a sleep channel. We can safely use the `intr_thread` structure itself as a sleep channel for an interrupt thread. This is very confusing.